OSSM-4835 [Doc] Capture Differences between OSSM 2.x and OSSM 3.0 (#84682)
openshift/openshift-docs: modules/ossm-migrating-read-me-sidecar-injection-considerations.adoc

{SMProductName} 2 supported using pod annotations and labels to configure sidecar injection and there was no need to indicate which control plane a workload belonged to.

With {SMProduct} 3, even though the Istio control plane discovers a namespace, the workloads present in that namespace still require sidecar proxies to be included as workloads in the service mesh, and to be able to use Istio's many features.

In {SMProduct} 3, sidecar injection works the same way as it does for Istio, with pod or namespace labels used to trigger sidecar injection. However, it might be necessary to include a label that indicates which control plane the workload belongs to.

[NOTE]
====
The Istio Project has deprecated pod annotations in favor of labels for sidecar injection.
====

When an `Istio` resource has the name `default` and `InPlace` upgrades are used, there is a single `IstioRevision` with the name `default` and the label `istio-injection=enabled` for sidecar injection.

However, an `IstioRevision` resource is required to have a different name in the following cases:

* Multiple control plane instances are present.
* A `RevisionBased`, canary-style control plane upgrade is in progress.

> If there are running multiple control plane instances, or you chose the `RevisionBased` update strategy during your {SMProduct} 3 installation, then an `IstioRevision` resource is required to have a different name than `default`. When that happens, it is necessary to use a label that indicates which control plane revision the workload(s) belong to by specifying `istio.io/rev=<istiorevision_name>`.

Reviewer: I think these two words might be switched:
```suggestion
If there are multiple running control plane instances, or you chose the `RevisionBased` update strategy during your {SMProduct} 3 installation, then an `IstioRevision` resource is required to have a different name than `default`. When that happens, it is necessary to use a label that indicates which control plane revision the workload(s) belong to by specifying `istio.io/rev=<istiorevision_name>`.
```

> If there are running multiple control plane instances, or you chose the `RevisionBased` update strategy during your {SMProduct} 3 installation, then an `IstioRevision` resource is required to have a different name than `default`. When that happens, it is necessary to use a label that indicates which control plane revision the workload(s) belong to by specifying `istio.io/rev=<istiorevision_name>`.

Reviewer: Also, per the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=punctuation-parentheses#plurals) we shouldn't use parentheses to indicate potential plurals:
```suggestion
If there are running multiple control plane instances, or you chose the `RevisionBased` update strategy during your {SMProduct} 3 installation, then an `IstioRevision` resource is required to have a different name than `default`. When that happens, it is necessary to use a label that indicates which control plane revision the workloads belong to by specifying `istio.io/rev=<istiorevision_name>`.
```
If you had to choose one and singular "workload" feels better, feel free to choose that option instead

These labels can be applied at the workload or namespace level.

You can inspect available revisions by running the following command:

> [source, terminal]

Reviewer: ```suggestion
[source,terminal]
```
